usaspending_api/accounts/v2/views/federal_account_program_activities_total.py

```diff
@@ -0,0 +1,86 @@
+from django.db.models import Case, Q, TextField, When, Value, Sum
+from django.db.models.functions import Coalesce
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from usaspending_api.accounts.v2.views.federal_account_base import FederalAccountBase, PaginationMixin
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+
+
+class FederalAccountProgramActivitiesTotal(PaginationMixin, FederalAccountBase):
+    """
+    Retrieve a list of program activity totals for a federal account.
+    """
+
+    endpoint_doc = (
+        "usaspending_api/api_contracts/contracts/v2/federal_accounts/federal_account_totals/program_activities.md"
+    )
+    filters: dict
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.default_sort_column = "obligations"
+        self.sortable_columns = ["obligations", "code", "name", "type"]
+
+    def post(self, request: Request, *args, **kwargs):
+        self.filters = request.data.get("filter", [])
+        query = self.get_filter_query()
+        results = (
+            FinancialAccountsByProgramActivityObjectClass.objects.filter(
+                Q(
+                    treasury_account__federal_account__federal_account_code=self.federal_account_code,
+                    submission__is_final_balances_for_fy=True,
+                )
+            )
+            .filter(query)
+            .values("submission_id")
```

Comment:
The query in close, but it shouldn't be grouping by the submission_id. You can actually have multiple annotation and values statements in order to have the ORM create exactly what you want.